control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -490,53 +497,61 @@ bool AEB::checkCollision(MarkerArray & debug_markers)
                                  ? hasCollision(current_v, closest_object_point.value())
                                  : false;
 
-    // Add debug markers
-    if (publish_debug_markers_) {
-      const auto [color_r, color_g, color_b, color_a] = debug_colors;
-      addMarker(
-        this->get_clock()->now(), path, ego_polys, objects, collision_data_keeper_.get(), color_r,
-        color_g, color_b, color_a, debug_ns, debug_markers);
-    }
+    time_keeper_->end_track("has_collision_with_" + debug_ns);
     // check collision using rss distance
     return has_collision;
   };
 
   // step3. make function to check collision with ego path created with sensor data
-  time_keeper_->start_track("has_collision_imu_path");
-  const auto has_collision_ego = [&](pcl::PointCloud<pcl::PointXYZ>::Ptr filtered_objects) -> bool {
+  const auto ego_imu_path = (!use_imu_path_ || !angular_velocity_ptr_)
+                              ? Path{}
+                              : generateEgoPath(current_v, angular_velocity_ptr_->z);
+
+  const auto ego_mpc_path = (!use_predicted_trajectory_ || !predicted_traj_ptr_)
+                              ? std::nullopt
+                              : generateEgoPath(*predicted_traj_ptr_);
+
+  PointCloud::Ptr filtered_objects = pcl::make_shared<PointCloud>();
+  if (use_pointcloud_data_) {
+    const std::vector<Path> paths = [&]() {
+      std::vector<Path> paths;
+      if (use_imu_path_) paths.push_back(ego_imu_path);
+      if (ego_mpc_path.has_value()) {
+        paths.push_back(ego_mpc_path.value());
+      }
+      return paths;
+    }();
```

Comment:
```suggestion
  const auto ego_imu_path = (use_imu_path_ && angular_velocity_ptr_)
                              ? generateEgoPath(current_v, angular_velocity_ptr_->z)
                              : Path{};

  const auto ego_mpc_path = (use_predicted_trajectory_ && predicted_traj_ptr_)
                              ? generateEgoPath(*predicted_traj_ptr_)
                              : std::nullopt;

  PointCloud::Ptr filtered_objects = pcl::make_shared<PointCloud>();
  if (use_pointcloud_data_) {
    std::vector<Path> paths;
    if (use_imu_path_) paths.push_back(ego_imu_path);
    if (ego_mpc_path.has_value()) paths.push_back(*ego_mpc_path);
```
For me suggested change looks simple and easy to understand but not strong opinion.